packages/sync/src/conflicts/conflictLink.ts

```diff
@@ -0,0 +1,51 @@
+import { ApolloLink } from "apollo-link";
+import { onError } from "apollo-link-error";
+import { GraphQLError } from "graphql";
+import { DataSyncConfig } from "../config/DataSyncConfig";
+import { ConflictResolutionData, strategies } from "./strategies";
+
+export class ConflictLink {
+    private config: DataSyncConfig;
+
+    constructor(config: DataSyncConfig) {
+        this.config = config;
+    }
+
+    /**
+     * Initialise the conflict link
+     * @returns An apollo link capable of detecting errors
+     */
+    public init(): ApolloLink {
+        const link = onError(({ graphQLErrors, operation, forward }) => {
+            const data = this.getConflictData(graphQLErrors);
+            if (!this.config.conflictStrategy) {
```

Comment:
I will extract that if to separate method that returns conflictStrategy function